src/sendqueue.rs

```diff
@@ -0,0 +1,92 @@
+use std::convert::TryInto;
+use std::ptr::NonNull;
+
+use libc::c_uint;
+
+use crate::raw;
+use crate::Error;
+use crate::{Active, Capture};
+
+pub struct SendQueue(NonNull<raw::pcap_send_queue>);
+
+pub enum Sync {
+    Off = 0,
+    On = 1,
+}
+
+impl SendQueue {
+    pub fn new(memsize: c_uint) -> Result<Self, Error> {
+        let squeue = unsafe { raw::pcap_sendqueue_alloc(memsize) };
+        let squeue = NonNull::new(squeue).ok_or(Error::InsufficientMemory)?;
+
+        Ok(Self(squeue))
+    }
+
+    pub fn maxlen(&self) -> c_uint {
+        unsafe { (*self.0.as_ptr()).maxlen }
+    }
+
+    pub fn len(&self) -> c_uint {
+        unsafe { (*self.0.as_ptr()).len }
```

Comment:
Aren't all your dereferincing calls like `(*self.0.as_ptr())` the same as `self.0.as_mut()`?